temporian/beam/io.py

```diff
@@ -0,0 +1,299 @@
+"""Utilities to import/export Beam-Event-Set from/to dataset containers."""
+
+from typing import Iterable, Dict, Any, Tuple, Union, Optional
+
+import csv
+import io
+import numpy as np
+import apache_beam as beam
+from apache_beam.io.fileio import MatchFiles
+from temporian.core.data.node import Schema
+from temporian.core.data.dtypes.dtype import DType
+from temporian.implementation.numpy.data.event_set import tp_dtype_to_np_dtype
+
+# Remark: We use tuples instead of dataclasses or named tuples as it seems
+# to be the most efficient solution for beam.
+
+# Type of an index element.
+#
+# In the numpy backend, index are represented as numpy primitives. However,
+# Beam does not support numpy primitive as index. Therefore, all index are
+# converted to python primitive of type "BeamIndex".
+BeamIndex = Union[int, float, str, bool]
+
+# Type of an index element OR a feature index (int).
+#
+# The feature index -1 is used to represent the timestamps of event-set without
+# features.
+BeamIndexAndFeature = BeamIndex
+
+# An array of timestamps.
+Timestamps = np.float64
+
+# A single event / row during the conversion from dict of key/value to internal
+# the Temporian beam format for event sets. In a StructuredRow, index and
+# features are indexed by integer instead of string keys.
+#
+# Contains: the index, the timestamps, and the features.
+# The indexes and features are ordered according to a Schema.
+# Note the double 2-items tuple (instead of a one 3-items tuple) to facilitate
+# Beam operations.
+StructuredRow = Tuple[
+    Tuple[BeamIndex, ...], Tuple[Timestamps, Tuple[np.generic, ...]]
+]
+
+# Unit of data for an event set used by all the operators implementations.
+#
+# Contains: the index+feature_idx, timestamps, and feature values.
+# The feature value can be None for event-set without features.
+BeamEventSet = Tuple[
```

Comment:
If I understand correctly, this isn't a whole EventSet, but a single index value? In that case, I would propose changing its name to `IndexValue` or a better one, and a `BeamEventSet` or `PEventSet` would be a `PCollection[IndexValue]` for example

Reply:
I see now that it's not a whole index value, but a single feature inside an index value (my comment still applies). Also: does this mean the timestamps are repeated for every feature an event set has?

Reply:
This new naming is much better.

>> Duplicated timestamps.
Yes, this is an important implication. Essentially, we are traddingof performance for scalability.
*Added a comment about the duplication of timestamps.